### PR TITLE
Time.ptp() method deprecated in astropy/time/core.py

### LINKED ISSUE #15379: Deprecate Time.ptp() method?
### What is the problem this feature will solve?

Numpy has ~deprecated~ removed the `ndarray.ptp()` method. This also means the `np.ptp()` function will no longer look for a `.ptp()` method, so will no longer work on `Time`. It seems to make sense to deprecate it on `Time` too (plain removal seems wrong; not sure why numpy didn't deprecate).

### Describe the desired outcome

The `Time` methods that are meant to mimic those of `ndarray` are kept up to date. We also ensure that corresponding numpy functions work on `Time` instances.

### Additional context

See #15378 for other updates related to the removal of `ndarray.ptp()` in numpy 2.0.

### PR DESCRIPTION
**Pull request by olintoeduardo**
### Description

This pull request is to address the issue #15379 to deprecate the Time.ptp() method following Numpy updates in version 2.0. It includes a deprecation warning in function ptp() inside astropy/time/core.py and adjustments in the astropy/time/tests/test_methods.py to reflect such changes. 

Fixes #15379

### CODE REVIEW

File: astropy/time/core.py
Comment (pllim) [inline astropy/time/core.py:1]: That is not how we handle deprecation in docstring. Please look at the other stuff that is already deprecated in this code base and follow a similar style. Thanks!

File: astropy/time/tests/test_methods.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 from astropy.time.utils import day_frac
 from astropy.units.quantity_helper.function_helpers import ARRAY_FUNCTION_ENABLED
 from astropy.utils import iers
+from warnings import warn
+from astropy.utils.exceptions import AstropyDeprecationWarning, AstropyWarning
+
```
Comment (pllim) [inline astropy/time/tests/test_methods.py:17]:
```suggestion
from astropy.utils.exceptions import AstropyDeprecationWarning
```

**Comment by neutrinoceros** [comment]
@olintoeduardo are you still interested in working on this ?

@pllim @mhvk this looks necessary for full Numpy 2.0 compat, isn't it ?

**Comment by pllim** [comment]
I thought @mhvk worked around the actual compat? Though this should happen too at some point, I think, but I will defer to Marten's expertise.

@neutrinoceros , if you want to wrap this up, best to open a new PR, as this PR needed some clean-up to be consistent with the rest of the codebase.

Thanks!

**Comment by neutrinoceros** [comment]
I'll wait for @mhvk's reply, but yeah, I'm happy to pick this up if needed !

**Pull request closed without merging**